validator/go.mod

```diff
@@ -0,0 +1,16 @@
+module otel_config_validator
+
+go 1.22.0
+
+require (
+	github.com/santhosh-tekuri/jsonschema/v5 v5.3.1
```

Comment:
@codeboten care to comment on the JSON schema tooling situation in the go ecosystem? I _believe_ the [go config implementation](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/config/generated_config.go#L1C22-L1C57) uses github.com/atombender/go-jsonschema

Reply:
Ah interesting. I'm curious to know too. I'll look to jog my memory on why I went with `jsonschema`.

Reply:
Oh, looks like its because github.com/santhosh-tekuri/jsonschema supports more validations. Maybe they are ones we don't need but I believe that is why I picked it.

Reply:
I think i ended up with go-jsonschema because it supports code generation from the schema, either should be fine as long as they support the use-case they need